### PR TITLE
Fix e2e tests

### LINKED ISSUE #505: Fix API E2E Tests
## Context
- A change in the schedule type a long time ago broke the e2e tests
- This wasn't caught cause we don't run these tests in our pipeline

## Todos
- Fix the e2e tests
- Add them to our CI pipeline

### PR DESCRIPTION
**Pull request by daisykucharski**
# Description

- Fixes multiple errors causing the e2e tests to fail (synchronize when in testing mode, ensuring testing data is correct, using right jest settings, ensuring we're querying for the right plans, etc.)
- Ran `yarn format` in api-v2 and as a result it has reformatted multiple unrelated files 

Closes #505

## Type of change

Please tick the boxes that best match your changes.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`
- [ ] This has migration changes and requires a run of `yarn dev:migration:run`

# How Has This Been Tested?

Ran tests locally and they are passing

# Checklist:

- [ ] I have run the production builds in docker for the frontend/backend and ensure things run fine. Check README of repo on how to run if not sure.
- [x] I have performed a self-review of my own code
- [x] I have commented my code where needed
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -31,4 +31,5 @@ Please describe how you tested this PR (both manually and with tests) Provide in
 - [ ] My changes generate no new warnings
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
+- [ ] I've run the end to end tests
```
Comment (BrandonLim8890) [inline .github/PULL_REQUEST_TEMPLATE:34]: NICE

File: packages/api-v2/test/major/major.e2e-spec.ts [hunk 2]
```diff
@@ -2,6 +2,9 @@ import { INestApplication } from "@nestjs/common";
 import * as request from "supertest";
 import { initializeApp } from "../../test/utils";
 
+jest.useRealTimers();
+jest.setTimeout(50000);
```
Comment (BrandonLim8890) [inline packages/api-v2/test/major/major.e2e-spec.ts:6]: Is there a reason why we include a timeout in the beginning of everything?

File: packages/api-v2/test/plan/plan.e2e-spec.ts [hunk 3]
```diff
@@ -96,7 +104,7 @@ describe("PlanController (e2e)", () => {
     await request(app.getHttpServer())
       .patch(`/plans/${planID}`)
       .set("Cookie", cookie)
-      .send({ catalogYear: 2018 })
+      .send({ name: "robert is stinky" })
```
Comment (BrandonLim8890) [inline packages/api-v2/test/plan/plan.e2e-spec.ts:107]: i like this

File: packages/api-v2/test/student/student.e2e-spec.ts [hunk 4]
```diff
@@ -26,12 +29,10 @@ describe("StudentController (e2e)", () => {
     uuid = res.body.uuid;
   });
 
-  afterEach(async () => {
-    await dropStudentTable(connection);
-  });
-
   afterAll(async () => {
```
Comment (BrandonLim8890) [inline packages/api-v2/test/student/student.e2e-spec.ts:32]: Hmm I'm not sure about having tests depend on each other. I'm assuming if we don't reset the user table after each test, we are persisting the environment throughout each test. Is it not better practice to reset the env before each test?

**Review by BrandonLim8890 — APPROVED** [review]
LOOKING CLEAN. Just have one question regarding whether we want to reset the entire testing env before each test vs persisting. Curious to hear your thoughts.

File: packages/api-v2/test/major/major.e2e-spec.ts (hunk 2 above)
Comment (daisykucharski) [reply packages/api-v2/test/major/major.e2e-spec.ts:6]: Works without it now, think there was a step in my debugging where I thought it was necessary lol

File: packages/api-v2/test/student/student.e2e-spec.ts (hunk 4 above)
Comment (daisykucharski) [reply packages/api-v2/test/student/student.e2e-spec.ts:32]: True that's better practice, but it doesn't work the other way and I don't know why lol

Comment (daisykucharski) [reply packages/api-v2/test/student/student.e2e-spec.ts:32]: Someone else in 3 semesters can figure it out

Comment (BrandonLim8890) [reply packages/api-v2/test/student/student.e2e-spec.ts:29]: BRUH LMAO ok i make a ticket

**Pull request merged**